### PR TITLE
[AOTInductor] Switch ProxyExecutor to use AtenTensorHandle 

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Summary: Switch ProxyExecutor to use AtenTensorHandle.

Test Plan: E2E Test

Differential Revision: D49471659

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @ngimel

### CODE REVIEW

File: torch/csrc/inductor/aoti_torch/shim_common.cpp [hunk 1]
```diff
@@ -147,6 +147,13 @@ AOTITorchError aoti_torch_empty_strided(
   });
 }
 
+AOTITorchError aoti_torch_new_tensor(AtenTensorHandle* ret) {
+  AOTI_TORCH_CONVERT_EXCEPTION_TO_ERROR_CODE({
+    at::Tensor* out_tensor = new at::Tensor();
```
Comment (yifuwang) [inline torch/csrc/inductor/aoti_torch/shim_common.cpp:152]: Who is responsible for deallocating this?

Comment (SherlockNoMad) [reply torch/csrc/inductor/aoti_torch/shim_common.cpp:152]: In the codege, there will be a `create_raii_tensor_handle_for_temp` call latter, which binds a deleter for this.

**Review by yifuwang — APPROVED** [review]

File: torch/csrc/inductor/aoti_torch/c/shim.h [hunk 2]
```diff
@@ -141,6 +141,11 @@ AOTI_TORCH_EXPORT AOTI_TORCH_NOINLINE AOTITorchError aoti_torch_empty_strided(
     int32_t device_type,
     int32_t device_index);
 
+// This function will create a new uninitialized tensor object
+// and its pointer is returned through *out.
+AOTI_TORCH_EXPORT AOTI_TORCH_NOINLINE AOTITorchError aoti_torch_new_tensor(
```
Comment (desertfire) [inline torch/csrc/inductor/aoti_torch/c/shim.h:146]: I suggest to add `uninitialized` in the name to make that explicit.

**Review by desertfire — APPROVED** [review]
LGTM. You need to rebase to the latest code, and adapt to the API changes in RAIIAtenTensorHandle.

**Review by chenyang78 — APPROVED** [review]
Not necessarily in this PR, but can we come up with a test. Particularly, since we changed  fill_output_arg to use correct APIs, we seems to be able to run some test with extern kernels?

File: torch/_inductor/codegen/wrapper.py [hunk 3]
```diff
@@ -1711,8 +1719,12 @@ def fill_args(arg, arg_type):
 
         def fill_output_arg(arg, return_type):
             if isinstance(return_type, torch.TensorType):
-                self.writeline(f"at::Tensor {arg};  // output buffer")
-                new_tensor_args.append(f"&{output_arg}")
+                self.writeline(f"AtenTensorHandle {arg}_handle;  // output buffer")
+                self.writeline(
+                    f"AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_new_uninitialized_tensor(&{arg}_handle));"
+                )
+                self.writeline(f"RAIIAtenTensorHandle {arg}({arg}_handle);")
```
Comment (chenyang78) [inline torch/_inductor/codegen/wrapper.py:1726]: Hmm, wondering if we need to guard these lines with ```config.aot_inductor.abi_compatible```?

File: torch/_inductor/codegen/wrapper.py [hunk 4]
```diff
@@ -1813,7 +1825,9 @@ def generate_extern_kernel_alloc_and_find_schema_if_needed_fbcode(
 
         tensor_args_var = f"tensor_args_var_{next(self.kernel_callsite_id)}"
         tensor_call_args_str = ", ".join(tensor_call_args)
-        self.writeline(f"void* {tensor_args_var}[] = {{{tensor_call_args_str}}};")
+        self.writeline(
+            f"AtenTensorHandle {tensor_args_var}[] = {{{tensor_call_args_str}}};"
```
Comment (chenyang78) [inline torch/_inductor/codegen/wrapper.py:1829]: Same here - do we need to guard ```AtenTensorHandle``` with ```config.aot_inductor.abi_compatible```?

File: torch/_inductor/codegen/wrapper.py (hunk 3 above)
Comment (SherlockNoMad) [reply torch/_inductor/codegen/wrapper.py:1726]: no, `generate_extern_kernel_args_decl_if_needed` is only used in the fbcode path, which always need abi=true.

File: torch/_inductor/codegen/wrapper.py (hunk 4 above)
Comment (SherlockNoMad) [reply torch/_inductor/codegen/wrapper.py:1829]: ditto. this is in fbcode path.

**Comment by huydhn** [comment]
@pytorchbot drci

(Please ignore this, I'm testing Dr.CI)

**Pull request closed without merging**